kernel/tests/read.rs

```diff
@@ -353,35 +356,159 @@ async fn stats() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
-macro_rules! assert_batches_sorted_eq {
-    ($expected_lines: expr, $CHUNKS: expr) => {
+macro_rules! sort_lines {
+    ($lines: expr) => {{
         // sort except for header + footer
-        let num_lines = $expected_lines.len();
+        let num_lines = $lines.len();
         if num_lines > 3 {
-            $expected_lines.as_mut_slice()[2..num_lines - 1].sort_unstable()
+            $lines.as_mut_slice()[2..num_lines - 1].sort_unstable()
         }
+    }};
+}
 
+// NB: expected_lines_sorted MUST be pre-sorted (via sort_lines!())
+macro_rules! assert_batches_sorted_eq {
+    ($expected_lines_sorted: expr, $CHUNKS: expr) => {
         let formatted = arrow::util::pretty::pretty_format_batches($CHUNKS)
             .unwrap()
             .to_string();
         // fix for windows: \r\n -->
-
         let mut actual_lines: Vec<&str> = formatted.trim().lines().collect();
-
-        // sort except for header + footer
-        let num_lines = actual_lines.len();
-        if num_lines > 3 {
-            actual_lines.as_mut_slice()[2..num_lines - 1].sort_unstable()
-        }
-
+        sort_lines!(actual_lines);
         assert_eq!(
-            $expected_lines, actual_lines,
+            $expected_lines_sorted, actual_lines,
             "\n\nexpected:\n\n{:#?}\nactual:\n\n{:#?}\n\n",
-            $expected_lines, actual_lines
+            $expected_lines_sorted, actual_lines
         );
```

Comment:
for the future: i'm wondering if we have a better way to do these assertions instead of string comparisons

Reply:
yeah.... maybe. the DAT tests do it a little differently, but end up having to do more complex `RecordBatch` sorting so they can be compared directly. Not 100% clear to me which is best yet.